### ISSUE TITLE
NPM version...

### ISSUE THREAD

**Issue by adamboutcher** [indico/newdle#339]
The documentation doesn't state what version of NPM is required, presumably it's newer than the defaul in CentOS7 due to the lack of npm ci in that packaged version.

**Comment by ThiefMaster** [comment]
Using the stable LTS provided by nodejs would be a good idea. :)

**Comment by adamboutcher** [comment]
There's just a general lack of system dependencies in general too...

**Comment by ThiefMaster** [comment]
I don't think we really have many system dependencies except maybe libpq-dev in order to install psycopg2 for database access.

**Comment by pferreir** [comment]
> I don't think we really have many system dependencies except maybe libpq-dev in order to install psycopg2 for database access.

uwsgi and nginx

**Comment by ThiefMaster** [comment]
Ah right, even though I see that as something the person installing a python webapp is responsible for, since in the end anything that speaks WSGI should work...

**Comment by adamboutcher** [comment]
There's literally no list of deps bar Python3.8 on the README. You cant get the web app running without

npm
PostgresSQL
uwsgi
httpd/nginx

Maybe it's worth having a list as we don't all use docker.

**Comment by pferreir** [comment]
> Ah right, even though I see that as something the person installing a python webapp is responsible for, since in the end anything that speaks WSGI should work...

Sure, but maybe we should probably help them by mentioning what they have to install and provide example config files? Other web apps normally provide such information in their standard documentation.

**Comment by adamboutcher** [comment]
`yum install -y centos-release-scl epel-release
yum groupinstall -y "development tools"
yum install -y rh-git218 rh-python38-python rh-python38-python-devel rh-python38-python-psycopg2 rh-nodejs12-npm rh-postgresql96 rh-postgresql96-postgresql-devel

source /opt/rh/rh-git218/enable
source /opt/rh/rh-nodejs12/enable
source /opt/rh/rh-python38/enable
source /opt/rh/rh-postgresql96/enable

git clone
cd newdle
make`

This is what I had to do to get a successful build.

**Comment by pferreir** [comment]
I don't think we should go as far as providing instructions for CentOS 7, but at least generic Debian and Fedora should be covered.

**Comment by adamboutcher** [comment]
Why not? If someone in the community is willing to provide details?
More people run CentOS as a server than Fedora.

**Comment by ThiefMaster** [comment]
It's something we would need to maintain eventually... and CentOS7 isn't even particularly new.

**Comment by adamboutcher** [comment]
CentOS is supported until 2024 and most science sites use CentOS, Fedora only has a lifetime of ~1 year. I get that the current CentOS support model for CentOS 8 isn't great but for a production deployment people want stability.

We run Indico on CentOS7, we plan to deploy this on a CentOS7 system (or CentOS 8 equivalent) ideally not in a container so we don't have to fight with docker every reboot for just a single service.

**Comment by pferreir** [comment]
I'd say it's not a problem if someone wants to take over the maintenance of CentOS-specific instructions. But I'd favor having a high-level dependency list which can be easily updated over that.
